components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.device.mgt.iot.common.api;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.policy.mgt.common.Policy;
+import org.wso2.carbon.policy.mgt.common.PolicyAdministratorPoint;
+import org.wso2.carbon.policy.mgt.common.PolicyManagementException;
+import org.wso2.carbon.policy.mgt.common.PolicyMonitoringTaskException;
+import org.wso2.carbon.policy.mgt.common.monitor.ComplianceData;
+import org.wso2.carbon.policy.mgt.common.monitor.PolicyComplianceException;
+import org.wso2.carbon.policy.mgt.core.PolicyManagerService;
+import org.wso2.carbon.policy.mgt.core.task.TaskScheduleService;
+
+import javax.jws.WebService;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+@WebService
+public class PolicyManagementService {
+
+	private static Log log = LogFactory.getLog(PolicyManagementService.class);
+
+	@Context  //injected response proxy supporting multiple thread
+	private HttpServletResponse response;
+
+	private PrivilegedCarbonContext ctx;
+
+	private PolicyManagerService getPolicyServiceProvider() throws DeviceManagementException {
+		String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+		PrivilegedCarbonContext.startTenantFlow();
+		ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+		ctx.setTenantDomain(tenantDomain, true);
+
+		if (log.isDebugEnabled()) {
+			log.debug("Getting thread local carbon context for tenant domain: " + tenantDomain);
+		}
+
+		PolicyManagerService policyManagerService = (PolicyManagerService) ctx.getOSGiService(
+				PolicyManagerService.class, null);
+
+		if (policyManagerService == null) {
+			String msg = "Policy Management service not initialized";
+			log.error(msg);
```

Comment:
this is fixed in [#90e57c7](https://github.com/rasika90/carbon-device-mgt/commit/90e57c70e092e87247183e42e8d44463a60091a2).